src/components/displayAllComponent.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import ChooseFolder from "./chooseFolder/chooseFolder";
+import { NavLink,Redirect } from "react-router-dom";
+import SvgSettingOptions from "./svgSetting/svgSettingOptions";
+import "../components/checkBoxSelection/checkBoxSelection.scss";
+import { changeObj, optimizeSvg } from "../../src/components/functions";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../store/actionTypes";
+
+class DisplayAllComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+     redirect : false
+    };
+  }
+  getOptimizeSvg = async () => {
+    for (var option of this.props.svgOptions) {
+      changeObj(option, this.props.svgObject.plugins);
+    }
+    for (var setting of this.props.svgSettingList) {
+      if (this.props.svgOptions.length > 0) {
+        this.props.displayOptimize.push({
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(
+            setting.dataurl,
+            this.props.svgObject.plugins
+          )
+        });
+      } else {
+        this.props.displayOptimize.push({
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(setting.dataurl, null)
+        });
+      }
+    }
+    this.setState({
+      redirect: true
+    })
```

Comment:
make this one line to increase readability please

Reply:
that makes sense for multiple states, but in this case it is just one, and because of this it makes it more difficult to read. In some cases we code in specific styles to accommodate the future but for others we use our discretion for what code should look like right now. Do you anticipate us setting a different state here in the near future?